docs/available-options.md
---
title: How it works - Storelocatorjs
description: Storelocatorjs includes multiple options to personnalize the map
---

Storelocatorjs contains many options to allow you to easily configure your map according to your needs. List of all options with their default values.

#### `apiKey`

`mandatory - default: string = ''`

Tells storelocatorjs the API key to use Google Maps API.

```js
new storelocatorjs({
  options: {
    apiKey: ''
  }
});
```

#### `webServiceUrl`

`mandatory - default: string = ''`

Tells storelocatorjs where is located the cloud function web service to get JSON store datas.

```js
new storelocatorjs({
  options: {
    webServiceUrl: ''
  }
});
```

#### `cluster.options`

`default: object`

Tells storelocatorjs marker clusters options. Documentation is available on [Google Developers](https://developers.google.com/maps/documentation/javascript/marker-clustering).

```js
new storelocatorjs({
    options: {
        cluster: {
            options: {
                {
                    averageCenter: true,
                    gridSize: 50,
                    imagePath: 'https://developers.google.com/maps/documentation/javascript/examples/markerclusterer/m',
                    maxZoom: 13,
                    minimumClusterSize: 2,
                    styles: [],
                    zoomOnClick: true
                }
            }
        }
    }
});
```

#### `cluster.status`

`default: boolean = false`

Tells storelocatorjs whether to enable the marker clustering.

```js
new storelocatorjs({
  options: {
    cluster: {
      status: false
    }
  }
});
```

#### `debug`

`default: boolean = false`

Tells storelocatorjs whether to display the debug overlay.

```js
new storelocatorjs({
  options: {
    debug: false
  }
});
```

#### `geolocation.startOnLoad`

`default: integer = false`

Tells storelocatorjs whether to enable an auto geolocation request on page load.

```js
new storelocatorjs({
  options: {
    geolocation: {
      startOnLoad: false
    }
  }
});
```

#### `geolocation.status`

`default: boolean = true`

Tells storelocatorjs whether to enable the geolocation.

```js
new storelocatorjs({
  options: {
    geolocation: {
      status: true
    }
  }
});
```

#### `map.markers.width`

`default: integer = 30`

Tells storelocatorjs what is the marker width in pixel (without unity).

```js
new storelocatorjs({
  options: {
    map: {
      markers: {
        width: 30
      }
    }
  }
});
```

#### `map.markers.height`

`default: integer = 40`

Tells storelocatorjs what is the marker height in pixel (without unity).

```js
new storelocatorjs({
  options: {
    map: {
      markers: {
        height: 40
      }
    }
  }
});
```

#### `map.markers.styles`

`default: array`

Tells storelocatorjs what are the geolocation marker styles. The category `userPosition` is reserved to identify the geolocation marker.

```js
new storelocatorjs({
  options: {
    map: {
      markers: {
        styles: [
          {
            category: 'userPosition',
            colorBackground: '#4285f4',
            colorText: '#fff'
          }
        ]
      }
    }
  }
});
```

The array `map.markers.styles` can accept entries for each category to customize markers by category.

```js
new storelocatorjs({
  options: {
    map: {
      markers: {
        styles: [
          {
            category: '1',
            colorBackground: '#ff6600',
            colorText: '#fff'
          },
          {
            category: '2',
            colorBackground: '#ffcc33',
            colorText: '#fff'
          },
          {
            category: '3',
            colorBackground: '#ea4c89',
            colorText: '#fff'
          }
        ]
      }
    }
  }
});
```

#### `map.options`

`default: object`

Tells storelocatorjs marker options. Documentation is available on [Google Developers](https://developers.google.com/maps/documentation/javascript/reference/map#MapOptions).

`map.options.styles` can be used to stylized the map. Documentation is available on the [feature page](https://storelocatorjs.github.io/features.html#maps-styles)

```js
new storelocatorjs({
  options: {
    map: {
      options: {
        center: [46.227638, 2.213749],
        disableDefaultUI: false,
        fullscreenControl: true,
        mapTypeControl: false,
        mapTypeId: 'roadmap',
        scaleControl: false,
        scrollwheel: true,
        streetViewControl: false,
        styles: [],
        zoom: 6
      }
    }
  }
});
```

#### `requests.searchRadius`

`default: integer = 50`

Tells storelocatorjs what is the search radius in kilometers, when a search is triggered by the user.

```js
new storelocatorjs({
  options: {
    requests: {
      searchRadius: 50
    }
  }
});
```

#### `requests.storesLimit`

`default: integer = 20`

Tells storelocatorjs what is the limit of results return by the request. Use `0` to return unlimited stores, according to the JSON database.

```js
new storelocatorjs({
  options: {
    requests: {
      storesLimit: 20
    }
  }
});
```

#### `selectors`

`default: object`

Tells storelocatorjs what are HTML identifier (class or data attribute) for each elements. You can personalize them.

```js
new storelocatorjs({
  options: {
    selectors: {
      container: '.storelocator',
      formSearch: '.storelocator-formSearch',
      geolocButton: '.storelocator-geolocButton',
      inputSearch: '.storelocator-inputSearch',
      loader: '.storelocator-loader',
      nav: '.storelocator-nav',
      searchFilters: '[data-filter]',
      sidebar: '.storelocator-sidebar',
      sidebarResults: '.storelocator-sidebarResults'
    }
  }
});
```

#### `markersUpdate.limitInViewport`

`default: integer = 30`

Tells storelocatorjs what is the maximum of results in the main viewport when an auto-search is triggered. Zoom or move events may trigger an auto-search if user extends the area.

```js
new storelocatorjs({
  options: {
    markersUpdate: {
      limitInViewport: 30
    }
  }
});
```

#### `markersUpdate.maxRadius`

`default: integer = 150`

Tells storelocatorjs what is the maximum search radius in kilometers during an auto-search.

```js
new storelocatorjs({
  options: {
    markersUpdate: {
      maxRadius: 150
    }
  }
});
```

#### `markersUpdate.status`

`default: boolean = true`

Tells storelocatorjs whether to enable auto-search to refresh markers on area bounds changed. Zoom or move events may trigger an auto-search if user extends the area.

```js
new storelocatorjs({
  options: {
    markersUpdate: {
      status: true
    }
  }
});
```

#### `markersUpdate.stepRadius`

`default: integer = 50`

Tells storelocatorjs what is the search radius when the default search limit of 30 kilometers is exceeded. Zoom or move events may trigger an auto-search if user extends the area.

```js
new storelocatorjs({
  options: {
    markersUpdate: {
      stepRadius: 50
    }
  }
});
```

<script>
  ((window.gitter = {}).chat = {}).options = {
    room: 'store-locator/store-locator'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>


src/demo/config.js
import './css/reset.css'
import './css/demo.css'
import 'storelocatorjs/dist/storelocator/css/storelocator.css'
import Storelocatorjs from 'storelocatorjs'

/* eslint-disable no-unused-vars */
const myStorelocator = new Storelocatorjs({
	options: {
		apiKey: 'AIzaSyD6vKbzlHT4wfk9jbCLXn_HZrdFaLzO2XI',
		webServiceUrl: 'https://storelocator-functions.vercel.app',
		cluster: {
			options: {
				averageCenter: true,
				gridSize: 50,
				imagePath:
					'https://developers.google.com/maps/documentation/javascript/examples/markerclusterer/m',
				maxZoom: 13,
				minimumClusterSize: 2,
				styles: [
					{
						url: 'images/cluster.png',
						textSize: 14,
						textColor: '#000',
						width: 70,
						height: 70
					}
				],
				zoomOnClick: true
			},
			status: true
		},
		geolocation: {
			startOnLoad: false,
			status: true
		},
		map: {
			markers: {
				width: 30,
				height: 40,
				styles: [
					{
						category: 'userPosition',
						colorBackground: '#33ccff',
						colorText: '#fff'
					},
					{
						category: '1',
						colorBackground: '#ff6600',
						colorText: '#fff'
					},
					{
						category: '2',
						colorBackground: '#ffcc33',
						colorText: '#fff'
					},
					{
						category: '3',
						colorBackground: '#ea4c89',
						colorText: '#fff'
					}
				]
			},
			options: {
				styles: [
					{
						featureType: 'administrative',
						elementType: 'labels.text.fill',
						stylers: [{ color: '#6195a0' }]
					},
					{
						featureType: 'administrative.province',
						elementType: 'geometry.stroke',
						stylers: [{ visibility: 'off' }]
					},
					{
						featureType: 'landscape',
						elementType: 'geometry',
						stylers: [{ lightness: '0' }, { saturation: '0' }, { color: '#f5f5f2' }, { gamma: '1' }]
					},
					{
						featureType: 'landscape.man_made',
						elementType: 'all',
						stylers: [{ lightness: '-3' }, { gamma: '1.00' }]
					},
					{
						featureType: 'landscape.natural.terrain',
						elementType: 'all',
						stylers: [{ visibility: 'off' }]
					},
					{
						featureType: 'poi',
						elementType: 'all',
						stylers: [{ visibility: 'off' }]
					},
					{
						featureType: 'poi.park',
						elementType: 'geometry.fill',
						stylers: [{ color: '#bae5ce' }, { visibility: 'on' }]
					},
					{
						featureType: 'road',
						elementType: 'all',
						stylers: [{ saturation: -100 }, { lightness: 45 }, { visibility: 'simplified' }]
					},
					{
						featureType: 'road.highway',
						elementType: 'all',
						stylers: [{ visibility: 'simplified' }]
					},
					{
						featureType: 'road.highway',
						elementType: 'geometry.fill',
						stylers: [{ color: '#fac9a9' }, { visibility: 'simplified' }]
					},
					{
						featureType: 'road.highway',
						elementType: 'labels.text',
						stylers: [{ color: '#4e4e4e' }]
					},
					{
						featureType: 'road.arterial',
						elementType: 'labels.text.fill',
						stylers: [{ color: '#787878' }]
					},
					{
						featureType: 'road.arterial',
						elementType: 'labels.icon',
						stylers: [{ visibility: 'off' }]
					},
					{
						featureType: 'transit',
						elementType: 'all',
						stylers: [{ visibility: 'simplified' }]
					},
					{
						featureType: 'transit.station.airport',
						elementType: 'labels.icon',
						stylers: [
							{ hue: '#0a00ff' },
							{ saturation: '-77' },
							{ gamma: '0.57' },
							{ lightness: '0' }
						]
					},
					{
						featureType: 'transit.station.rail',
						elementType: 'labels.text.fill',
						stylers: [{ color: '#43321e' }]
					},
					{
						featureType: 'transit.station.rail',
						elementType: 'labels.icon',
						stylers: [
							{ hue: '#ff6c00' },
							{ lightness: '4' },
							{ gamma: '0.75' },
							{ saturation: '-68' }
						]
					},
					{
						featureType: 'water',
						elementType: 'all',
						stylers: [{ color: '#eaf6f8' }, { visibility: 'on' }]
					},
					{
						featureType: 'water',
						elementType: 'geometry.fill',
						stylers: [{ color: '#c7eced' }]
					},
					{
						featureType: 'water',
						elementType: 'labels.text.fill',
						stylers: [{ lightness: '-49' }, { saturation: '-53' }, { gamma: '0.79' }]
					}
				]
			}
		}
	},
	onReady: function (map) {
		// this.triggerRequest({
		// 	lat: 48.8589507,
		// 	lng: 2.2770202
		// })
	}
})
